src/_gettsim/groupings.py

```diff
@@ -0,0 +1,125 @@
+from typing import Callable
+
+import numpy
+
+
+def create_groupings() -> dict[str, Callable]:
+    return {"bg_id": bg_id_numpy, "st_id": st_id_numpy}
+
+
+def bg_id_numpy(  # noqa: PLR0913
+    p_id: numpy.ndarray,
+    hh_id: numpy.ndarray,
+    alter: numpy.ndarray,
+    p_id_ehepartner: numpy.ndarray,
```

Comment:
```suggestion
    p_id_einstandspartner: numpy.ndarray,
```
Being married is not required, but relationship has to be close to it ("eheähnliche Gemeinschaft")